contracts/validator_manager.v.py

```diff
@@ -128,49 +147,78 @@ def get_validators_max_index() -> int128:
     return activate_validator_num + self.empty_slots_stack_top
 
 
-# Adds a validator to the validator set, with the validator's size being the msg.value
-# (ie. amount of ETH deposited) in the function call. Returns the validator index.
+# Helper functions to get notary info in notary_registry
+@public
+@constant
+def get_notary_info(notary_address: address) -> (int128, int128):
+    return (self.notary_registry[notary_address].deregistered, self.notary_registry[notary_address].pool_index)
+
+
+# Adds an entry to notary_registry, updates the notary pool (notary_pool, notary_pool_len, etc.),
+# locks a deposit of size NOTARY_DEPOSIT, and returns True on success.
 @public
 @payable
-def deposit() -> int128:
-    validator_addr: address = msg.sender
-    assert not self.is_validator_deposited[validator_addr]
-    assert msg.value == self.deposit_size
-    # find the empty slot index in validators set
-    index: int128 = self.num_validators
-    if not self.is_stack_empty():
-        index = self.stack_pop()        
-    self.validators[index] = {
-        deposit: msg.value,
-        addr: validator_addr,
+def register_notary() -> bool:
+    assert msg.value == self.NOTARY_DEPOSIT
```

Comment:
assert msg.value **>=** self.NOTARY_DEPOSIT

Reply:
I was wondering if it's really necessary to allow deposit more than required?

Reply:
I was wondering if it's really necessary to forbid deposit more than required? :P